src/OrchardCore.Modules/OrchardCore.Twitter/Migrations/TwitterMigrations.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using Dapper;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Data;
+using OrchardCore.Data.Migration;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Descriptor.Models;
+using OrchardCore.Environment.Shell.Scope;
+using YesSql;
+
+namespace OrchardCore.Twitter.Migrations;
+
+public class TwitterMigrations : DataMigration
+{
+    private readonly ShellDescriptor _shellDescriptor;
+
+    public TwitterMigrations(ShellDescriptor shellDescriptor)
+    {
+        _shellDescriptor = shellDescriptor;
+    }
+
+    public int Create()
```

Comment:
After using https://github.com/OrchardCMS/OrchardCore/pull/16350 to be able to add an `UpdateTwitterStatusTask`, I've switched to this branch to test the migrations. However, I still get:

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/02c5abec-ed59-4928-a9fe-e6f7a7345d39)

Reply:
Thanks @Piedone for the App_Data snapshot. The root cause was the Migration was skipped because it was wrapped in a call to `_shellDescriptor.WasFeatureAlreadyInstalled("OrchardCore.Twitter")` which returns false. I expected that this method would return true when the feature was installed but something related to the feature's serial number matching the shellDescriptor's serial number causes it to return false. The [code is here](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore.Abstractions/Shell/Extensions/ShellDescriptorExtensions.cs#L19).

I removed the call to that `WasFeatureAlreadyInstalled` method and the migration succeeded.

Reply:
Ok, I pushed the latest changes to fix this and tested against your App_Data snapshot, @Piedone. Looks to work as expected now.
![image](https://github.com/user-attachments/assets/18580a49-18ec-44b5-835b-93518491873a)